### DISCUSSION TITLE
quickly delete entities

### DISCUSSION THREAD

**Discussion by SheetMetalPerfo** [haplokuon/netDxf#416] [Q&A]
Hello. Thank you for this great tool. I'm new to C# and object programming.
My goal is to delete a large number of entities as quickly as possible. These are circles. There are about 300,000 to remove. But it could be much more.
To delete the entities I use:
loaded.RemoveEntity(circlesToDel) where circlesToDel is List<Circle>
and it takes a long time.
What is the most efficient way to delete entities?
If you have a code snippet that would be great.
Thanks

**Comment by haplokuon** [comment]
The first thing you should do is download the latest source code and compile it, use the release build the debug will be slower. Try it and check if there is an improvement. Removing so many entities in one go can take some time, there is some clean up that has to be done when one entity is deleted, and that takes time. I run a test on may system i5 - 6500 and removing 300,000 lines took about 45 seconds, most of the time was spent removing its reference from other dictionaries. At the moment it is what it is and I cannot do much about it.

If you need to remove so many entities, perhaps it will be faster to clone the ones you want to keep and add them to a new document.

**Comment by SheetMetalPerfo** [comment]
Hello,
Thank you for your reply.
It's true that I only use debug mode to test my routines.
I will take your advice and see if it improves the execution time.
Thank you again for your great tool...
